ios/RNSModalScreen.mm

```diff
@@ -7,6 +7,26 @@
 
 @implementation RNSModalScreen
 
+// When using UIModalPresentationStyleFullScreen the views belonging to the presenting view controller are removed,
+// Thus there is no RCTRootView mounted and no trait observer working.
+// We have to set up an extra observer and send change notifications accoridngly
+// For more information, see https://github.com/software-mansion/react-native-screens/pull/2211.
```

Comment:
```suggestion
// When using UIModalPresentationStyleFullScreen the whole view hierarchy mounted under primary `UITransitionView` is removed, including React's root view, which observes for trait collection changes & sends it to `Appearance` module via system notification centre. To workaround this detached-root-view-situation we emit the event to React's `Appearance` ourselves.
// For more information, see https://github.com/software-mansion/react-native-screens/pull/2211.
```

Reply:
Let's also add a link to lines from `RCTRootView` file in react native's repository, from where most of this code is taken.

Reply:
[chore: updated comment](https://github.com/software-mansion/react-native-screens/pull/2211/commits/3283fe3aa82504c08449be7abf27df5b0c733512)